Tasks/TaskModel.swift

```diff
@@ -0,0 +1,69 @@
+///
+//  TaskModel.swift
+//  Tasks
+//
+//  Created by Maximilian Alexander on 8/26/21.
+//
+//  Copyright © 2023 DittoLive Incorporated. All rights reserved.
+
+import DittoSwift
+
+struct TaskModel {
+    let _id: String
+    let body: String
+    let userId: String
+    let isCompleted: Bool
+    let isSafeForEviction: Bool
+    let invitationIds: [String: Bool]
+    
+    /// Returns instance from `QueryResultItem.value
+    init(_ value: [String: Any?]) {
+        self._id = value["_id"] as! String
+        self.body = value["body"] as! String
+        self.userId = value["userId"] as? String ?? ""
+        self.isCompleted = value["isCompleted"] as? Bool ?? false
+        self.isSafeForEviction = value["isSafeForEviction"] as? Bool ?? false
+        self.invitationIds = value["invitationIds"] as? [String:Bool] ?? [:]
+    }
+}
+
+extension TaskModel: Codable {
+    private static let decoder = JSONDecoder()
+    
+    /// Returns optional instance decoded from `QueryResultItem.jsonString()`
+    static func withJson(_ json: String) -> TaskModel? {
+        try? decoder.decode(Self.self, from: Data(json.utf8))
+    }
```

Comment:
It could be nicer as a tutorial to handle thrown errors since getting errors from decoding JSONs is very common. Below is an example with `init?(json:)`:

```suggestion
    init?(json: String) {
        do {
            self = try JSONDecoder().decode(Self.self, from: Data(json.utf8))
        } catch {
            print("ERROR:", error.localizedDescription)
            return nil
        }
    }
```